subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java

```diff
@@ -349,6 +353,10 @@ private DefaultJavaCompileSpec createBaseSpec() {
     private void applyToolchain(ForkOptions forkOptions) {
         final JavaInstallationMetadata metadata = getToolchain();
         forkOptions.setJavaHome(metadata.getInstallationPath().getAsFile());
+        if (isToolchainCompatibleWithJava16()) {
+            // used by the incremental compiler plugin, see org.gradle.internal.compiler.java.ClassNameCollector
+            forkOptions.getJvmArgs().addAll(Arrays.asList("--add-opens", "jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED"));
```

Comment:
I might be worthing looking to actually fixing this without the add opens. I pushed a small WIP here that I played around with, we might not need this workaround for this. WDYT?

https://github.com/gradle/gradle/commit/2fc4346101a1f717728ee1a57ba9b0f2412718ec